sympy/simplify/besselsimp.py

```diff
@@ -0,0 +1,97 @@
+from __future__ import print_function, division
+
+from sympy.core.numbers import I, pi
+from sympy.core import Mul, S, Dummy
+from sympy.functions import exp_polar, exp, log
+from sympy.functions.special.bessel import besselj, besseli, besselk, jn, bessely
+from sympy.simplify.trigsimp import trigsimp, exptrigsimp
+
+
+def besselsimp(expr):
+    """
+    Simplify bessel-type functions.
+
+    This routine tries to simplify bessel-type functions. Currently it only
+    works on the Bessel J and I functions, however. It works by looking at all
+    such functions in turn, and eliminating factors of "I" and "-1" (actually
+    their polar equivalents) in front of the argument. Then, functions of
+    half-integer order are rewritten using strigonometric functions and
+    functions of integer order (> 1) are rewritten using functions
+    of low order.  Finally, if the expression was changed, compute
+    factorization of the result with factor().
+
+    >>> from sympy import besselj, besseli, besselsimp, polar_lift, I, S
+    >>> from sympy.abc import z, nu
+    >>> besselsimp(besselj(nu, z*polar_lift(-1)))
+    exp(I*pi*nu)*besselj(nu, z)
+    >>> besselsimp(besseli(nu, z*polar_lift(-I)))
+    exp(-I*pi*nu/2)*besselj(nu, z)
+    >>> besselsimp(besseli(S(-1)/2, z))
+    sqrt(2)*cosh(z)/(sqrt(pi)*sqrt(z))
+    >>> besselsimp(z*besseli(0, z) + z*(besseli(2, z))/2 + besseli(1, z))
+    3*z*besseli(0, z)/2
+    """
+    # TODO
+    # - better algorithm?
+    # - simplify (cos(pi*b)*besselj(b,z) - besselj(-b,z))/sin(pi*b) ...
+    # - use contiguity relations?
+
+    from .simplify import unpolarify
```

Comment:
I think that ``unpolarify`` and ``polarify`` do not fit well in the category of simplifications. They might be moved over possibly to ``complexes`` which seems to contain most of the methods related to polar numbers. Since all of ``functions`` is imported before ``simplify``, no importation problem should remain.